models/marts/dim_contract_code_hist.sql

```diff
@@ -0,0 +1,242 @@
+{{
+    config(
+        materialized = 'incremental',
+        unique_key = ['ledger_key_hash', 'start_date'],
+        cluster_by = ["ledger_key_hash", "start_date", "row_hash"],
+        tags = ["soroban_analytics", "dimension", "scd-2", "daily"]
+    )
+}}
+
+/*
+Model: dim_contract_code_hist
+
+Description:
+------------
+This model implements Slowly Changing Dimension (SCD) Type 2 logic to track historical changes
+in contract code data over time. The model supports initial loads and incremental updates, including catch-up.
+
+Features:
+---------
+1. Creates a record for each change in `ledger_key_hash`.
+2. Applies change tracking logic using row hash comparison to detect updates.
+3. Implements date chaining to maintain historical records, marking `is_current` for the latest active record.
+4. Supports multi-day catch-up scenarios by processing all relevant records based on the incremental logic.
+
+Usage:
+------
+Compile:
+    dbt compile --models dim_contract_code_hist
+
+Run:
+    dbt run --models dim_contract_code_hist --full-refresh
+    dbt run --models dim_contract_code_hist
+*/
+
+-- Load latest changes from transformed contract code data, applying incremental filter
+with
+    source_data as (
+        select
+            contract_code_hash
+            , ledger_key_hash
+            , n_instructions
+            , n_functions
+            , n_globals
+            , n_table_entries
+            , n_types
+            , n_data_segments
+            , n_elem_segments
+            , n_imports
+            , n_exports
+            , n_data_segment_bytes
+            , contract_create_ts
+            , contract_delete_ts
+            , closed_at
+            , date(closed_at) as start_date
+            , airflow_start_ts
+            , batch_id
+            , batch_run_date
+            , row_hash
+        from {{ ref('int_transform_contract_code') }}
+        {% if is_incremental() %}
+            where closed_at > (select coalesce(max(closed_at), '2000-01-01T00:00:00+00:00') from {{ this }})
```

Comment:
nit: I don't think the coalesce is necessary. On the first run of the model `is_incremental()` will be false anyways and on subsequent runs the table will have data

Reply:
- This will help if we ever perform a Manual Truncate and don't want to modify anything in the airflow vars to trigger a Full refresh.

Reply:
When would you do a manual truncate without just running `--full_refresh`?

Reply:
If we ever want to do a clean up by Truncating the table i guess.
 
For full refresh, we have to make changes to airflow vars as its stands (my understanding based on existing setup?)
If we allow this code, we can simulate a refresh (assuming no schema changes) without having to do anything in airflow.

I am open to either options I guess as a Truncate table is also a manual step, a more straight forward thing compared to airflow changes. We can discuss in office hours